runtime/tchannel_client.go

```diff
@@ -268,7 +268,12 @@ func (c *tchannelOutboundCall) finish(err error) {
 
 	// emit metrics
 	if err != nil {
-		c.metrics.SystemErrors.Inc(1)
+		tcause := tchannel.GetSystemErrorCode(errors.Cause(err))
+		if terr, ok := c.metrics.SystemErrors[byte(tcause)]; ok {
```

Comment:
nit: you should define `c.metrics.SystemErrors` as a typedef like

```
type SystemErrorMap map[tchannel.SystemErrorCode]tally.Counter
```

Then you can define helper function like

```
func (m SystemErrorMap) Incr(cause tchannel.SystemErrorCode) {
    counter, ok := m[cause]
    if !ok {
        m[0x00].Inc(1)
        return
    }
    counter.Inc(1)
}
```

And this code changes to simply

```
c.metrics.SystemErrors.Incr(tcause)
```

And you can delete the duplicate code from below.

Reply:
Do you mind we do a style clean also for http status metrics at once in a separate diff? following the same style here but I think your comment is solid

I'm trying not to over-DRY here, the metrics logic between `tchannel_client.go` and `tchannel_server.go` seems a bit repetitive, but the metric behavior for endpoint & client might not always be the same (tchannel server might not need to emit timeout at all)

Reply:
OK